client/www/components/dash/explorer/EditRowDialog.tsx

```diff
@@ -45,6 +45,12 @@ function tryJsonParse(value: any) {
 }
 
 function getAppropriateFieldType(attr: SchemaAttr, value: any): FieldType {
+  if (!value && attr.checkedDataType) {
```

Comment:
Not required for this PR, just realized, these checks could be rewritten as something like: 

if (value) {
   ...
}
if (attr.checkedDataType) { 
  ...
} 
if (attr.inferredTyped?.length) { 
 ...

Reply:
Updated!

Also, I think this fixes a potential bug where `value` is `0`, `false`, or some other falsey thing.